### ISSUE TITLE
borg truncates last cmdline argument

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#2140]
from the borgbackup mailing list:

this is a strange issue - i get this only on one system (yet). (SLES 12 SP2)

I'm using borg 1.0.9, the "all in one" binary from the github page and it seems to cut off the last letter from the hostname...

```
# borg init ssh://testname
Remote: ssh: Could not resolve hostname testnam: Name or service not known
```
only borg seems to do this (ssh straight to the machine works from the same prompt).

adding a space to the end fixes the name: (well obviously testname doesn't exist, but it's looking up the right name now)
```
# borg init "ssh://testname "
Remote: ssh: Could not resolve hostname testname: Name or service not known
```

**Comment by ThomasWaldmann** [comment]
This could be a pyinstaller bug maybe (off-by-one error).

Just strange that this issue never surfaced before.
The normal release binaries run through the usual testsuite and such an error would be noticed there.

Advised to test with 1.0.10rc1 binaries, including the pyi321-debug one.

**Comment by ThomasWaldmann** [comment]
reply on the mailing list:

> How is that system different from your other systems (on which you use the same binary)?

this is the first SLES12SP2 System I tried. The others were CentOS, Ubuntu and Debian.

> > I'm using borg 1.0.9, the "all in one" binary from the github page and it seems to cut off the last letter from the hostname...
> Give the sha256sum of the binary you use, please.

```
# sha256sum /usr/local/bin/borg-linux64-1.0.9
9fa23310aa8b08a5d7427970250d9ec0f9512b03bbac32659079df8ae1031764  /usr/local/bin/borg-linux64-1.0.9
```
there's a symlink to this binary in `/usr/local/bin/borg`.

> That might be a pyinstaller bootloader issue. Can you try 1.0.10rc1 ?

tried and that one's better:
```
# borg-linux64-1.0.10rc1 init ssh://testmachine
usage: borg-linux64-1.0.10rc1 init [-h] [--critical] [--error] [--warning]
                                   [--info] [--debug] [--lock-wait N]
                                   [--show-rc] [--no-files-cache] [--umask M]
                                   [--remote-path PATH]
                                   [-e {none,keyfile,repokey}] [-a]
                                   [REPOSITORY]
borg-linux64-1.0.10rc1 init: error: argument REPOSITORY: Invalid location format: "ssh://testmachine"
```

> Try the normal linux64 binary as well as the pyi321-debug one.

The py321-debug one does it right too.

> Sounds like an off-by-one error. Just strange this did not surface before.

Yes - I'm also confused - I used borg on quite a lot of systems and this is the first that's having this problem. Maybe some environment variable influencing this that's only set on this system? I didn't find a clue yet...

**Comment by ThomasWaldmann** [comment]
Try this with the 1.0.9 binary:

```
borg init ssh://hostname/repopath
```

I suspect this is the sloppy location parsing regex in 1.0.9 which was fixed in 1.0.10rc1 (and **not** pyinstaller).

**Comment by aklaver** [comment]
Tested with 1.0.10rc1, without specifying repo name to replicate original post:

aklaver@killi:~> ./borg-linux64_1010rc1 init --remote-path /home/aklaver/bin/borg_new ssh://arkansas
usage: borg-linux64_1010rc1 init [-h] [--critical] [--error] [--warning]
                                 [--info] [--debug] [--lock-wait N]
                                 [--show-rc] [--no-files-cache] [--umask M]
                                 [--remote-path PATH]
                                 [-e {none,keyfile,repokey}] [-a]
                                 [REPOSITORY]
borg-linux64_1010rc1 init: error: argument REPOSITORY: Invalid location format: "ssh://arkansas"
 
So the parsing seems to be fixed in 1.0.10.

**Comment by ThomasWaldmann** [comment]
OK, thanks for testing this.

Closing as already fixed. 1.0.10 will get released soon (sunday? early next week?).
